python/cudf/cudf/utils/cudautils.py

```diff
@@ -553,7 +553,7 @@ def gpu_unique_set_insert(vset, sz, val):
 
 
 @cuda.jit
-def gpu_shift(in_col, out_col, N):
+def gpu_shift(in_col, out_col, N, replacement_value):
```

Comment:
Per our discussion, let's explicitly refer to this as a sentinel value so it's clear that this is not intended to be used generally for replacement.

Reply:
Any reason we can't just handle nulls using the utility functions now? We have functions to get and set mask bits.